src/keyword_vpd_parser.cpp

```diff
@@ -133,4 +133,91 @@ void KeywordVpdParser::checkNextBytesValidity(uint8_t i_numberOfBytes)
     }
 }
 
+types::DbusVariantType KeywordVpdParser::readKeywordFromHardware(
+    const types::ReadVpdParams i_paramsToReadData)
+{
+    types::Keyword l_keyword;
+
+    if (const types::Keyword* l_kwData =
+            std::get_if<types::Keyword>(&i_paramsToReadData))
+    {
+        l_keyword = *l_kwData;
+    }
+    else
+    {
+        logging::logMessage("Given VPD type is not supported.");
+        throw types::DbusInvalidArgument();
+    }
+
+    // Iterate through VPD vector to find the keyword
+    if (findKeyword(l_keyword) != 0)
+    {
+        logging::logMessage("Keyword " + l_keyword + " not found.");
+        throw types::DbusInvalidArgument();
+    }
+
+    // Skip bytes representing the keyword name
+    std::advance(m_vpdIterator, constants::TWO_BYTES);
+
+    // Get size of the keyword
+    const auto l_keywordSize = *m_vpdIterator;
+
+    // Skip bytes representing the size of the keyword
+    std::advance(m_vpdIterator, constants::ONE_BYTE);
+
+    // Read the keyword's value and return
+    return types::DbusVariantType{types::BinaryVector(
```

Comment:
why?

Reply:
do the operation and send the final value in the return statement, Dont do multiple things in return statement.
This is the review comment got it for my PR.

What I realised from this is
1. It avoids any possible error calculations made at return statement.
2. For developer testing or in future code debugging, you can add print statement to print the value for debugging.
3. in case we need logging, we can easily add without rewriting the return statement.

Reply:
1. It avoids any possible error calculations made at return statement.
it's developer's responsibility to check on the calculations. if there are any errors we can fix it with defect. Having a separate variable doesn't guarantee that the calculation will be error free.

2. For developer testing or in future code debugging, you can add print statement to print the value for debugging.
Code doesn't stop you to add print statements for debugging.

3. in case we need logging, we can easily add without rewriting the return statement.
Now in this case is it required to log a message with variant<vector<uint8_t> contents? 

In my case, I don't need to construct an object for std::vector and std::variant just to return it back to the caller. In place construction optimises the memory usage.